src/main/kotlin/org/danilopianini/plagiarismdetector/repository/content/RepoContentSupplierCloneStrategy.kt

```diff
@@ -0,0 +1,42 @@
+package org.danilopianini.plagiarismdetector.repository.content
+
+import org.apache.commons.io.FileUtils
+import org.apache.commons.io.filefilter.DirectoryFileFilter
+import org.apache.commons.io.filefilter.SuffixFileFilter
+import org.eclipse.jgit.api.Git
+import java.io.File
+import java.net.URL
+import java.nio.file.Files
+
+/**
+ * A strategy which locally clones the repository directory to supply its content.
+ * @property cloneUrl the repository [URL] to clone.
+ */
+class RepoContentSupplierCloneStrategy(private val cloneUrl: URL) : RepoContentSupplierStrategy {
+    companion object {
+        private const val URL_SEPARATOR = "/"
+    }
+    private val clonedRepoDirectory: File = cloneRepo()
+
+    private fun cloneRepo(): File {
+        val tmpDir = Files.createTempDirectory(cloneUrl.path.substringAfterLast(URL_SEPARATOR)).toFile()
```

Comment:
We might want to cache this at some point...

Reply:
sorry, could you explain me better?

Reply:
Sure: right now it is not essential, but, in actual execution, we might run the detector on multiple projects, and thus we probably want to cache the information we extract from the repos (if not the entire repos).

Reply:
Yes, I thought we would cache the extracted information from sources, like the token strings, but of course, we can cache also the entire repo content.

Reply:
It may make more sense. It's just a "let's keep that in mind".